### PR TITLE
Bumps versions and fixes compilation issues

### PR DESCRIPTION
**Pull request by JoseAlcerreca**

### CODE REVIEW

File: BasicRxJavaSample/app/build.gradle [hunk 1]
```diff
@@ -2,11 +2,10 @@ apply plugin: 'com.android.application'
 
 android {
     compileSdkVersion build_versions.compile_sdk
-    buildToolsVersion build_versions.build_tools
 
     defaultConfig {
         applicationId "com.example.android.persistence"
-        minSdkVersion build_versions.min_sdk
+        minSdkVersion 21
```
Comment (manuelvicnt) [inline BasicRxJavaSample/app/build.gradle:8]: Nit: In other samples, this variable is in the `versions.gradle` file

File: BasicRxJavaSampleKotlin/app/build.gradle [hunk 2]
```diff
@@ -21,11 +21,10 @@ apply plugin: 'kotlin-android-extensions'
 
 android {
     compileSdkVersion build_versions.compile_sdk
-    buildToolsVersion build_versions.build_tools
 
     defaultConfig {
         applicationId "com.example.android.observability"
-        minSdkVersion build_versions.min_sdk
+        minSdkVersion 21
```
Comment (manuelvicnt) [inline BasicRxJavaSampleKotlin/app/build.gradle:27]: `build_versions.min_sdk` exists in the `versions.gradle` file

File: MADSkillsNavigationSample/app/build.gradle [hunk 3]
```diff
@@ -21,7 +21,7 @@ apply plugin: "androidx.navigation.safeargs.kotlin"
 apply plugin: 'kotlin-kapt'
 
 android {
-    compileSdkVersion 29
+    compileSdkVersion 31
```
Comment (manuelvicnt) [inline MADSkillsNavigationSample/app/build.gradle:24]: Nit: you can use the versions in `versions.gradle`

File: NavigationAdvancedSample/app/src/main/res/values/colors.xml [hunk 4]
```diff
@@ -19,4 +19,5 @@
     <color name="colorPrimary">#008577</color>
     <color name="colorPrimaryDark">#00574B</color>
     <color name="colorAccent">#7e53c5</color>
+
```
Comment (manuelvicnt) [inline NavigationAdvancedSample/app/src/main/res/values/colors.xml:22]: Nit: this can be reverted

**Review by manuelvicnt — COMMENTED** [review]
LG overall. Some comments can be applied to all samples.

The PR is missing a yaml file for the `MADSkillsNavigationSample` project

File: .github/ci-gradle.properties [hunk 5]
```diff
@@ -0,0 +1,23 @@
+#
+# Copyright 2020 The Android Open Source Project
```
[the remark this answers is not in the document]
Comment (JoseAlcerreca) [reply .github/ci-gradle.properties:2]: Let's not address nits that would make this PR even bigger

File: BasicRxJavaSample/versions.gradle [hunk 6]
```diff
@@ -21,55 +21,55 @@
  **/
 ext.deps = [:]
 def versions = [:]
-versions.activity = '1.1.0'
-versions.android_gradle_plugin = '4.0.0'
-versions.annotations = "1.0.0"
-versions.apache_commons = "2.5"
-versions.appcompat = "1.2.0-alpha02"
+versions.activity = '1.6.1'
+versions.android_gradle_plugin = '7.4.0-rc03'
```
[the remark this answers is not in the document]
Comment (JoseAlcerreca) [reply BasicRxJavaSample/versions.gradle:25]: I think an RC is stable enough

File: BasicRxJavaSample/app/build.gradle (hunk 1 above)
Comment (JoseAlcerreca) [reply BasicRxJavaSample/app/build.gradle:8]: Yes, this is WAI for this particular sample.

File: BasicRxJavaSampleKotlin/app/build.gradle (hunk 2 above)
Comment (JoseAlcerreca) [reply BasicRxJavaSampleKotlin/app/build.gradle:27]: min_sdk is quite low so we only use it in some samples. This one breaks with that.

File: MADSkillsNavigationSample/app/build.gradle (hunk 3 above)
Comment (JoseAlcerreca) [reply MADSkillsNavigationSample/app/build.gradle:24]: We don't own this sample and it doesn't build so my plan is to deprecate it, so I didn't want to change too much.

File: BasicRxJavaSample/versions.gradle (hunk 6 above)
[the remark this answers is not in the document]
Comment (manuelvicnt) [reply BasicRxJavaSample/versions.gradle:25]: Not particularly worried about the stability but more about how accessible the version is. Most people will have the latest stable version already installed or it's easy to get it. Getting the RC version of Android Studio is more difficult in general as you don't see it in the Android studio landing page

**Review by tunjid — APPROVED** [review]
LGTM

**Pull request merged**